pkg/operator/baremetal_test.go

```
@@ -0,0 +1,78 @@
+	}
+}
+
+func newOperatorWithBaremetalConfig() *OperatorConfig {
+	return &OperatorConfig{
+		targetNamespace,
+		Controllers{
+			"docker.io/openshift/origin-aws-machine-controllers:v4.0.0",
+			"docker.io/openshift/origin-machine-api-operator:v4.0.0",
+			"docker.io/openshift/origin-machine-api-operator:v4.0.0",
+		},
+		BaremetalControllers{
+			"quay.io/openshift/origin-baremetal-operator:v4.2.0",
+			"quay.io/openshift/origin-ironic:v4.2.0",
+			"quay.io/openshift/origin-ironic-inspector:v4.2.0",
+			"quay.io/openshift/origin-ironic-ipa-downloader:v4.2.0",
+			"quay.io/openshift/origin-ironic-rhcos-downloader:v4.2.0",
+			"quay.io/openshift/origin-ironic-static-ip-manager:v4.2.0",
+		},
+	}
+}
+
+//Testing the case where the password does already exist
+func TestCreateMariadbPasswordSecret(t *testing.T) {
+	kubeClient := fakekube.NewSimpleClientset(nil...)
+	operatorConfig := newOperatorWithBaremetalConfig()
+	client := kubeClient.CoreV1()
+
+	// First create a mariadb password secret
+	if err := createMariadbPasswordSecret(kubeClient.CoreV1(), operatorConfig); err != nil {
+		t.Errorf("Failed to create new Mariadb password. %s ", err)
+		return
+	}
+	// Make sure mariadb password exists.
+	oldMaridbPassword, _ := client.Secrets(operatorConfig.TargetNamespace).Get(baremetalSecretName, metav1.GetOptions{})
+	oldPassword, ok := oldMaridbPassword.StringData["password"]
+	if !ok {
+		t.Fatal("Failure extracting first Mariadb password from Secret.")
+	}
+	if oldPassword == "" {
+		t.Errorf("Error reading first Mariadb password from Secret.")
+	} else {
+		t.Logf("First Mariadb password : %s", oldPassword)
+	}
+
+	// The pasword definitely exists. Try creating again.
+	if err := createMariadbPasswordSecret(kubeClient.CoreV1(), operatorConfig); err != nil {
+		t.Fatal("Failure creating second Mariadb password.")
+	}
+	newMaridbPassword, _ := client.Secrets(operatorConfig.TargetNamespace).Get(baremetalSecretName, metav1.GetOptions{})
```

as above: why are we ignoring error here?